tests/types/implicitly_convertible.cmake

```diff
@@ -299,4 +299,40 @@ function("${test_cpp_implicitly_convertible}")
         cpp_implicitly_convertible(result bool double)
         ct_assert_equal(result FALSE)
     endfunction()
+
+    ct_add_section(NAME "pointer_to_desc")
+    function("${pointer_to_desc}")
+        cpp_implicitly_convertible(result bool* desc)
+        ct_assert_equal(result TRUE)
+    endfunction()
+
+    ct_add_section(NAME "double_pointer_to_desc")
+    function("${double_pointer_to_desc}")
+        cpp_implicitly_convertible(result bool** desc)
+        ct_assert_equal(result TRUE)
+    endfunction()
+
+    ct_add_section(NAME "pointer_to_pointer")
+    function("${pointer_to_pointer}")
+        cpp_implicitly_convertible(result bool* bool*)
+        ct_assert_equal(result TRUE)
+    endfunction()
+
+    ct_add_section(NAME "double_pointer_to_double_pointer")
+    function("${double_pointer_to_double_pointer}")
+        cpp_implicitly_convertible(result bool** bool**)
+        ct_assert_equal(result TRUE)
+    endfunction()
+
+    ct_add_section(NAME "desc_to_pointer")
+    function("${desc_to_pointer}")
+        cpp_implicitly_convertible(result desc bool*)
+        ct_assert_equal(result TRUE)
+    endfunction()
+
+    ct_add_section(NAME "pointer_to_bool")
+    function("${pointer_to_bool}")
+        cpp_implicitly_convertible(result bool* bool)
+        ct_assert_equal(result FALSE)
+    endfunction()
```

Comment:
Maybe add another test to make sure a double pointer is not implicitly convertible to a single pointer as well.

Reply:
Added